edx_rest_framework_extensions/auth/jwt/middleware.py

```diff
@@ -72,19 +86,87 @@ def _add_missing_jwt_permission_classes(self, view_class):
             view_class.permission_classes += tuple(classes_to_add)
 
     def process_view(self, request, view_func, view_args, view_kwargs):  # pylint: disable=unused-argument
-        # Views as functions store the view's class in the 'view_class' attribute.
-        # Viewsets store the view's class in the 'cls' attribute.
-        view_class = getattr(
-            view_func,
-            'view_class',
-            getattr(view_func, 'cls', view_func),
-        )
+        view_class = _get_view_class(view_func)
 
         view_authentication_classes = getattr(view_class, 'authentication_classes', tuple())
-        if self._includes_base_class(view_authentication_classes, BaseJSONWebTokenAuthentication):
+        if _includes_base_class(view_authentication_classes, BaseJSONWebTokenAuthentication):
             self._add_missing_jwt_permission_classes(view_class)
 
 
+class JwtAuthWithLoginRequiredMiddleware(MiddlewareMixin):
+    """
+    Middleware enables DRF JwtAuthentication authentication class for endpoints
+    using the LoginRequired permission class.
+
+    Enables a non-standard use of DRF, because DRF only supports returning
```

Comment:
Does it make sense to classify this as 'non-standard'?

Reply:
It's non-standard when it comes to DRF.  That is why we need special middleware.

Do you have a proposal that is better and strong enough?  I think this is the right solution in certain situations, but not many.

Reply:
See separate rephrasing comment.